memcached/memcached.go

```diff
@@ -23,47 +63,80 @@ func NewMemCachedClient(serverList []string) (*CacheClient, error) {
 	return c, nil
 }
 
-// getItem retrieves an Item from cache
-// It returns (nil, err) if it is not able to retrieve the item, else returns (Item,nil)
-func (c *CacheClient) getItem(key string) (*memcache.Item, error) {
+// GetItem takes in the key, expiration and a dbCallBack function.
+// If a cache miss occurs, the dbCallBack function is called which retrieves data from the database.
+// This value from the database is saved back to memcache.
+// expiration is the cache expiration time, in seconds: either a relative
+// time from now (up to 1 month), or an absolute Unix epoch time.
+// Zero means the Item has no expiration time.
+// It returns (nil, err) if there's any other error, else returns an interface{} object.
+func (c *CacheClient) GetItem(key string, expiration int32, dbCallBack func() (interface{}, error)) (interface{}, error) {
```

Comment:
here the dbcallback func() has a empty parameter list. Will it not be limited?

Reply:
It will be wrapped inside an anonymous function